### DISCUSSION TITLE
Multiline folding

### DISCUSSION THREAD

**Discussion by denisborovikov** [moalamri/vscode-inline-fold#69] [General]
Hey, thanks for the amazing plugin!

I'm trying to use it to fold the `sx` prop from the [MUI]( https://mui.com/system/getting-started/the-sx-prop/ ) using these settings 

`sx={{(\s*.*?\s*)}}`
`gs`
`1`

<img src=https://user-images.githubusercontent.com/831065/202009635-e6df296e-06cf-4734-854f-0849dc4fbbb3.png width=400 />

It works great with one-line props but with multiline, the plugin hides the code but doesn't fold lines. Is is by design or am I doing it wrong?

<img src=https://user-images.githubusercontent.com/831065/202010273-f6552d1e-932c-45fe-8e8f-7c4531876d2a.png width=400 />

**Comment by moalamri** [comment]
Hi @denisborovikov 

It's neither by design or that you are doing something wrong, 😄 
As I have commented before:
> The extension can't fold multiline because it's basically a decorator not a real API (VS Code doesn't support inline-folding and it's been in the backlog for 4 years).
If you were to fold multiline with this extension then you will have empty lines due to being decorated to be hidden and not actually folded 😞.

**Reply by denisborovikov** [reply]
Gotcha, thank you for the reply. It's a useful plugin, anyway. Thank you!

**Comment by benjamin-kirkbride** [comment]
> it's been in the backlog for 4 years

Can we get a link to the upstream VSCode issue?

**Reply by moalamri** [reply]
It is 5 years now
https://github.com/microsoft/vscode/issues/50840

**Reply by benjamin-kirkbride** [reply]
> Yes, this can only be implemented in core, but there's currently no such work planned.

:disappointed:

**Comment by LeoGytis** [comment]
I use both tailwind and MUI so this is the regex implemented:
	**"inlineFold.regex": "(sx={{[^{}]*}}|className=\"[^\"]*\")",**

Still waiting when it would fold multiple lines
